### PR TITLE
Dark mode: Add `$theme-colors-dark` and `$theme-colors-rgb-dark`

### PR DESCRIPTION
**Pull request by louismaximepiton**
### Description

Description of the main branch once merged:

Palette:
| Variable to use 🎨 | Light mode hex | Dark mode hex |
| :------------------ | :---------------: | :---------------: |
| `--bs-body-bg` | `#ffffff` | `#141414` |
| `--bs-body-color` | `#000000` | `#eeeeee` |
| `--bs-highlight-bg` | `#000000` | `#ffffff` |
| `--bs-highlight-color` | `#ffffff` | `#000000` |
| `--bs-secondary-bg` | `#eeeeee` | `#333333` |
| `--bs-border-color-translucent` | `#cccccc` | `#666666` |
| `--bs-secondary-color` | `#666666` | `#cccccc` |
| `--bs-link-hover-color` | `#f16e00` | `#ff7900` |
| `--bs-primary` | `#ff7900` | `#ff7900` |
| `--bs-success` | `#228722` | `#32c832` |
| `--bs-info` | `#4170d8` | `#527edb` |
| `--bs-warning` | `#ffcc00` | `#ffcc00` |
| `--bs-danger` | `#cd3c14` | `#d53f15` |

If there's a new color that doesn't fit well in this table and none exist in actual, introduce a new one otherwise, update this description.

### Tasks

- [x] Display `theme-colors-dark-map` in the documentation (and `theme-colors-rgb-dark`?) where `theme-colors-map` is displayed **Should be fine**
- [x] Do we provide something to explain how to override `$theme-colors-dark` in Boosted only (doesn't exist in Bootstrap) **Should be fine**
- [ ] In `docs/5.3/utilities/background/`, it shows that we need to add a new text utility: white in light mode, black in dark mode.
- [ ] Does `bg-primary` need to switch between oranges in light and dark mode?
- [ ] Does `bg-dark` need to be `#141414`? And in light mode which color? (see `site/data/theme-colors.yml` and `contrast_color`)
- [ ] Refactoring of https://boosted.netlify.app/docs/5.3/utilities/colors/#oranges-colors so that it works in light/dark mode
- [ ] https://deploy-preview-2227--boosted.netlify.app/docs/5.3/customize/color/#oranges-colors switch light/dark mode or always display the 2 palettes?
- [x] Check `$body-secondary-color-dark` impacts in terms of usage **Should be fine**
- [x] Check `$body-secondary-bg-dark` impacts in terms of usage **Should be fine**
- [ ] Check `$body-tertiary-color-dark` impacts in terms of usage **Not quite sure, need further investigation**
- [x] Check `$body-tertiary-bg-dark` impacts in terms of usage **Should be fine**

### Types of change

- New feature (non-breaking change which adds functionality)

### Live previews

- <https://deploy-preview-2227--boosted.netlify.app/>

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -43,9 +43,9 @@ $grays: (
 $accessible-orange: #f16e00 !default;
 $brand-orange:      #ff7900 !default;
 //// Functional colors
-$functional-green:  #32c832 !default;
-$functional-blue:   #527edb !default;
-$functional-yellow: #fc0 !default;
+$functional-green:  #228722 !default;
+$functional-blue:   #4170d8 !default;
+$functional-yellow: #8f7200 !default;
```
Comment (louismaximepiton) [inline scss/_variables.scss:48]: Could be temporary. Kept it just to see where it's implemented and where it's not.

**Comment by louismaximepiton** [comment]
TBH, not quite sure about the changes in the doc. maybe these should come later. Haven't changed the migration guide as weel for now since the variables might change.

File: scss/_maps.scss [hunk 2]
```diff
@@ -57,11 +57,30 @@ $background-colors: map-merge(
...
 // End mod
 
+$theme-colors-dark: null !default;
+$theme-colors-rgb-dark: null !default;
 $theme-colors-text-dark: null !default;
 $theme-colors-bg-subtle-dark: null !default;
 $theme-colors-border-subtle-dark: null !default;
 
 @if $enable-dark-mode {
+  // scss-docs-start theme-color-dark-map
```
Comment (julien-deramond) [inline scss/_maps.scss:67]: Missing Boosted mods in this file

File: scss/_maps.scss [hunk 3]
```diff
@@ -57,11 +57,30 @@ $background-colors: map-merge(
...
+  $theme-colors-dark: (
```
Comment (julien-deramond) [inline scss/_maps.scss:68]: `$theme-colors` is defined in `_variables.scss`. If we want to respect a certain logic, is it possible to define it in `_variables-dark.scss` instead?

Note: `$theme-colors-rgb-dark` can stay there.

File: scss/_root.scss [hunk 4]
```diff
@@ -207,6 +207,14 @@
     --#{$prefix}tertiary-bg: #{$body-tertiary-bg-dark};
     --#{$prefix}tertiary-bg-rgb: #{to-rgb($body-tertiary-bg-dark)};
 
+    @each $color, $value in $theme-colors-dark {
```
Comment (julien-deramond) [inline scss/_root.scss:210]: Missing Boosted mods

**Review by julien-deramond — COMMENTED** [review]
Few quick comments

File: scss/_variables-dark.scss [hunk 5]
```diff
@@ -6,46 +6,81 @@
...
+$functional-yellow-dark: $functional-yellow !default;
+$functional-red-dark:    #d53f15 !default;
+// scss-docs-end brand-colors-dark
+// End mod
+
 // scss-docs-start sass-dark-mode-vars
+// scss-docs-start theme-color-dark-variables
+$primary-dark:                      $orange !default;
+$secondary-dark:                    $black !default;
+$success-dark:                      $functional-green-dark !default;
```
Comment (julien-deramond) [inline scss/_variables-dark.scss:22]: Don't we need to override `$blue`, `$green`, etc. + `$colors` or create `*-dark` versions of them?

In light mode we have:

```scss
$functional-green:  #228722 !default;
$green: $functional-green !default;
$success: $green !default;
```

In dark mode:

```scss
$functional-green-dark:  #228722 !default;
$success: $functional-green-dark !default;
```

**Pull request merged**